### PR TITLE
bar generation max vs actual vs min

### PR DESCRIPTION
**Pull request by BainanXia**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Integrate another plot function from spot check integration notebook, which addresses #263 : bar plot max generation vs actual generation vs minimum generation for any type of resource in a scenario. 

### What the code is doing
- Implement plot function `plot_bar_generation_max_min_actual`.
- Add demo notebook for the new plot function.

### Testing
The plot function calls existing functions in `analyze` module for calculations.

### Where to look
```
/postreise/plot/plot_bar_generation_max_min_actual.py
/postreise/plot/demo/bar_generation_max_min_actual_demo.ipynb
```

### Usage Example/Visuals
Demos can be found in the demo notebook.

### Time estimate
15 min.

### CODE REVIEW

File: postreise/plot/plot_bar_generation_max_min_actual.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+def plot_bar_generation_max_min_actual(
+    scenario,
+    interconnect,
+    gen_type,
+    percentage=False,
+    show_as_state=True,
+    fontsize=15,
+    plot_show=True,
+):
+    """Generate for a given scenario the bar plot of maximum generation vs. actual
```
Comment (rouille) [inline postreise/plot/plot_bar_generation_max_min_actual.py:20]: Should we use total capacity in place of maximum generation?

File: postreise/plot/plot_bar_generation_max_min_actual.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    generation vs. minimum generation for a specific type of generators in an
+    interconnection.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance.
+    :param str interconnect: the interconnection name of interest.
+    :param str gen_type: type of generator.
+    :param bool percentage: show bars in percentage of maximum generation or not,
```
Comment (rouille) [inline postreise/plot/plot_bar_generation_max_min_actual.py:27]: same comment here

File: postreise/plot/plot_bar_generation_max_min_actual.py (hunk 1 above)
Comment (BainanXia) [reply postreise/plot/plot_bar_generation_max_min_actual.py]: Done.

File: postreise/plot/plot_bar_generation_max_min_actual.py (hunk 2 above)
Comment (BainanXia) [reply postreise/plot/plot_bar_generation_max_min_actual.py]: Done.

**Review by rouille — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**